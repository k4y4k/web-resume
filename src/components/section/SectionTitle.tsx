import "twin.macro";
import * as React from "react";

interface SectionTitleTypes {
  title: string;
  onCoverLetter: boolean;
}

const SectionTitle = ({
  title,
  onCoverLetter,
}: SectionTitleTypes): JSX.Element | null => {
  if (title === "") return null;

  if (onCoverLetter) return null;

  return (
    <h2
      data-testid="sectionTitle"
      tw="font-mono text-lg font-bold uppercase text-orchid-700"
    >
      {title}
    </h2>
  );
};

export default SectionTitle;
