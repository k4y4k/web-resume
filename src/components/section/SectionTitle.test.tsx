import * as React from 'react'
import { render, screen } from '@testing-library/react'
import SectionTitle from './SectionTitle'

describe('<SectionTitle />', () => {
  test('no data? no wakkas', () => {
    render(<SectionTitle title='' />)

    const title = screen.queryByTestId('sectionTitle')
    expect(title).not.toBeInTheDocument()
    expect(title).toMatchSnapshot()
  })

  test('displays title', () => {
    render(<SectionTitle title='Check this' />)

    const title = screen.getByTestId('sectionTitle')
    expect(title).toHaveTextContent(/check this/i)
    expect(title).toMatchSnapshot()
  })
})
