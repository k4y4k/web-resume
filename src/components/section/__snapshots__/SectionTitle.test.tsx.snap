// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionTitle /> displays title 1`] = `
.c0 {
  font-family: Inconsolata,monospace;
  font-size: 1.125rem;
  line-height: 1.75rem;
  font-weight: 700;
  text-transform: uppercase;
  --tw-text-opacity: 1;
  color: rgba(77,54,140,var(--tw-text-opacity));
}

<h2
  class="c0"
  data-testid="sectionTitle"
>
  Check this
</h2>
`;

exports[`<SectionTitle /> does not display title on the cover page 1`] = `<div />`;

exports[`<SectionTitle /> no data? no wakkas 1`] = `null`;
